### PR TITLE
docs: add prop types

### PR DESCRIPTION
**Pull request by mattrosno**
Adds prop types to all pages and components to help document our code and also proactively catch type errors while developing.

#### Changelog

**New**

- Prop types to all page and components

**Changed**

- Spread props in the `RequireAuth` component please review @francinelucca 

**Removed**

- The `sunset` value in JSDoc typedefs, as that's no longer part of the schema

#### Testing / reviewing

Run the web app and watch the console and terminal for prop type warnings. The only warning should be from Carbon's `AspectRatio` component that is missing some values that it supports.

### CODE REVIEW

File: services/web-app/components/catalog-results/catalog-results.js [hunk 1]
```diff
@@ -22,4 +23,8 @@ const CatalogResults = ({ assets = [] }) => {
   )
 }
 
+CatalogResults.propTypes = {
+  assets: PropTypes.array
```
Comment (mattrosno) [inline services/web-app/components/catalog-results/catalog-results.js:27]: Fixing this to be an array of assets.

Comment (mattrosno) [reply services/web-app/components/catalog-results/catalog-results.js]: Done.

File: services/web-app/components/auth/require-auth.js [hunk 2]
```diff
@@ -4,12 +4,20 @@
...
  * LICENSE file in the root directory of this source tree.
  */
-const RequireAuth = (props) => {
-  if (props.isAuthorized) {
-    return { ...props.children }
+import PropTypes from 'prop-types'
+
+const RequireAuth = ({ children, fallback: Fallback, isAuthorized }) => {
+  if (isAuthorized) {
+    return { children }
```
Comment (francinelucca) [inline services/web-app/components/auth/require-auth.js:11]:
```suggestion
    return children
```

File: services/web-app/components/auth/require-auth.js [hunk 3]
```diff
@@ -4,12 +4,20 @@
...
   } else {
-    return <props.fallback />
+    return <Fallback />
   }
 }
 
+RequireAuth.propTypes = {
+  children: PropTypes.node,
+  fallback: PropTypes.element,
```
Comment (francinelucca) [inline services/web-app/components/auth/require-auth.js:19]:
```suggestion
  fallback: PropTypes.func,
```

**Review by francinelucca — CHANGES_REQUESTED** [review]

**Comment by mattrosno** [comment]
@francinelucca thanks for the review. I didn't exactly know how to test that auth component, so I guessed that fallback prop would be of type "element" guessing it was a React component passed in.

Docs: https://www.npmjs.com/package/prop-types

```javascript
  // A React element (ie. <MyComponent />).
  optionalElement: PropTypes.element,
```

**Comment by francinelucca** [comment]
> @francinelucca thanks for the review. I didn't exactly know how to test that auth component, so I guessed that fallback prop would be of type "element" guessing it was a React component passed in.
> 
> Docs: https://www.npmjs.com/package/prop-types
> 
> ```js
>   // A React element (ie. <MyComponent />).
>   optionalElement: PropTypes.element,
> ```

Yeah I thought so too but was giving me an error something like `Invalid prop "Fallback" of type "function" supplied to "RequireAuth", expected a single ReactElement.`
I believe because we're passing a component type `FourOhFour` instead of a component instance `<FourOhFour />`. Anyways, I tested it with Proptypes.func and the warning is gone, everything else LGTM!
image

**Review by francinelucca — APPROVED** [review]

**Comment by mattrosno** [comment]
@francinelucca got it. Thanks for the thorough review!

**Pull request merged**